test/presubmit-tests.sh

```diff
@@ -31,6 +31,29 @@ CODE_PACKAGES_STR="${CODE_PACKAGES[*]}"
 CODE_PACKAGES_STR="./${CODE_PACKAGES_STR// /\/... .\/}/..."
 readonly CODE_PACKAGES_STR
 
+# Check for subdirs which do not contain any test module but contain code.
+# This is to prevent an issue where these directories are skipped in coverage
+# reports resulting in inaccurately high coverage results. Fix by adding a
+# stub_tests.go with no tests to these directories.
+# This function namespaced with eventing to allow adding to test-infra without name collision
+function check_eventing_subdirs_without_tests() {
```

Comment:
I think this kind of check is too hard. You're checking all directories and failing the build test job if there are directories without tests, even if they're unrelated to the current PR. This, at least, add a burden to PRs unrelated to these empty directories. Also, you're now enforcing a code coverage blocker, and as such I think it should be (1) agreed by the leads (instead of being just decided in a PR) and (2) only imposed after knative/eventing is in a better coverage state (so the burden is lower).

Reply:
By the way, if the goal is just to get accurate test coverage and the number is wrong because of missing test files, the code coverage tool should be responsible for dealing that, not the repo or the developers. @steuhs is the right point of contact about the coverage tool.

Reply:
Makes sense. Please merge https://github.com/knative/test-infra/pull/134 first and update the vendored test-infra to use the helper method.